tests/sentry/snuba/metrics/test_mqb_query_transformer.py

```diff
@@ -0,0 +1,1321 @@
+import datetime
+import re
+
+import pytest
+from snuba_sdk.aliased_expression import AliasedExpression
+from snuba_sdk.column import Column
+from snuba_sdk.conditions import BooleanCondition, BooleanOp, Condition, Op
+from snuba_sdk.entity import Entity
+from snuba_sdk.expressions import Granularity, Limit, Offset
+from snuba_sdk.function import Function
+from snuba_sdk.orderby import Direction, OrderBy
+from snuba_sdk.query import Query
+
+from sentry.snuba.metrics import MetricConditionField, MetricField, MetricGroupByField, MetricsQuery
+from sentry.snuba.metrics import OrderBy as MetricsOrderBy
+from sentry.snuba.metrics.mqb_query_transformer import (
+    MQBQueryTransformationException,
+    tranform_mqb_query_to_metrics_query,
+)
+
+"""
+Notes:
+- Parameter argument order for non column arguments should match the order of arguments after (aggregate_filter,
+org_id) or just (aggregate_filter) in the SnQL generator functions. As an example, if we define a histogram
+function in SnQL as:-
+Function(
+    function="histogram",
+    parameters=[
+        Column("d:transactions/duration@millisecond"),
+        0,  # histogram_from
+        5,  # histogram_to
+        5,  # num_buckets
+    ],
+    alias="histogram_transaction_duration",
+)
+
+then the order of arguments after the Column argument (i.e. numeric arguments) should be histogram_from, histogram_to,
+num_buckets matching the order in the following SnQL generator function definition
+
+def histogram_snql_factory(aggregate_filter, histogram_from, histogram_to, histogram_buckets, alias)
+
+These SnQL function generators can be found in `sentry/src/sentry/snuba/metrics/fields/base.py`, and these are
+defined for all derived operations i.e. operations not supported by clickhouse like rate, count_web_vitals,
+histogram (as it requires extra logic over the clickhouse function supported by datasketch), team_key_transaction,
+and count_transaction_name (for unparameterized and None)
+
+All non clickhouse functions (derived functions) are listed here
+https://github.com/getsentry/sentry/blob/0eb312411989c2dbde3cf0b5094d47829d01c854/src/sentry/snuba/metrics/fields/base.py#L1505-L1540
```

Comment:
```suggestion
https://github.com/getsentry/sentry/blob/0eb312411989c2dbde3cf0b5094d47829d01c854/src/sentry/snuba/metrics/fields/base.py#L1505-L1540
```
Wdyt to also add or replace with:

`src/sentry/snuba/metrics/fields/base.py:: DERIVED_OPS`

we did similar in other places, it's just easy to read.

Reply:
Yeah I think it is a good idea